rescript/ncbi.py

```diff
@@ -177,11 +332,39 @@ def get_nuc_for_accs(accs, entrez_delay=0.):
     return seqs, taxids
 
 
-def get_nuc_for_query(query, entrez_delay=0.):
+def _get_query_chunk(
+        chunk, params, entrez_delay, expected_num_records, request_lock,
+        logging_level):
+    params['retstart'] = chunk
+    data_chunk = _efetch_5000(params, request_lock, logging_level,
+                              entrez_delay)
+    if len(data_chunk) != min(5000, expected_num_records-chunk):
+        raise RuntimeError('Download did not finish. Reason unknown.')
+    logger = _get_logger(logging_level)
+    logger.info('got ' + str(len(data_chunk)) + ' sequences')
+    return data_chunk
+
+
+def get_nuc_for_query(
+        query, logging_level, n_jobs, request_lock, entrez_delay=0.334):
     params = dict(
-        db='nuccore', term=query, usehistory='y', retmax=0
+        db='nuccore', term=query, usehistory='y', retmax=0, **_entrez_params
     )
-    records = _get(params, entrez_delay=entrez_delay)
+    params, expected_num_records = _esearch(params, logging_level,
+                                            entrez_delay)
+    if expected_num_records > 166666:
+        _large_warning(logging_level)
+    records = []
+    logger = _get_logger(logging_level)
+    logger.info('Downloading ' + str(expected_num_records) + ' sequences')
+
+    parallel = Parallel(n_jobs=n_jobs, backend='loky')
```

Comment:
Even though there's no upper limit on the number of jobs that a user can request, max sensible `n_jobs` is a function of `expected_num_records` and `retmax=5000`. At the same time, a user may naively set `n_jobs` to a very high number what would result in creating a lot of (potentially) unnecessary processes. Can we limit the number of `n_jobs` here to what actually is required based on those two parameters in case user provided an unreasonable number?

Reply:
We don't know `expected_num_records` when the user uses a query rather than requesting a list of accession ids until we're well inside the program.

We could put an arbitrary upper bound on this to stop people from crashing their machines but I imagine that anyone who does that will learn fairly quickly that requesting 500 processes doesn't work, and we're unlikely to get a help request from them :)

The help string for `n-jobs` is already

> Number of concurrent download connections. More is faster until you run out of bandwidth.

Is that enough of a warning?

Reply:
Haha, you have a point. Also, clearly I tricked myself into thinking we always know both numbers - my bad. Yeah, I think the warning is good 👍